src/pass/plan_memory.cc

```diff
@@ -98,6 +108,21 @@ class GraphAllocator {
       num_match_color_ = pass::ColorNodeGroup(
           *idx_, importance, num_match_color_, &node_color_);
     }
+    // Init the free memory pool
+    if (shared_pool != nullptr) {
```

Comment:
consider another strategy where you don't insert shared pool to free memory, instead you run planing separately, sort both shared_pool and new pool, then match from small to big.